### PR TITLE
DRAFT: detect/bsize: Convert bsize into depth

### PR DESCRIPTION
**Pull request by jlucovsky**
DRAFT

If the conditions are correct, convert `bsize` into `depth` or `endswith`.

This change is for prototype eval only and doesn't consider the case of the bsize value being incompatible with content needs.

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [ ] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [ ] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
-
-
-

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 20362

**Comment by catenacyber** [comment]
Why is CI so red ?
What do you expect out of this draft ?

File: src/detect-bsize.c [hunk 1]
```diff
@@ -186,6 +188,7 @@ static int SigParseGetMaxBsize(const DetectU64Data *bsz)
     SCReturnInt(-1);
 }
 
+
```
Comment (jufajardini) [inline src/detect-bsize.c:191]: nit unnecessary new line?

**Comment by catenacyber** [comment]
Closing as stale, feel free to reopen Jeff and provide more details

**Pull request closed without merging**